### PR TITLE
Initial experiment 3

### PR DESCRIPTION
**Pull request by HGinnerup**

### CODE REVIEW

**Review by jamadaha — APPROVED** [review]

**Review by jamadaha — CHANGES_REQUESTED** [review]
Crashes on run

File: Jantimizer/Histograms/Managers/MinDepthHistogramManager.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            histogram.GenerateHistogramFromSortedGroups(await DataGatherer.GetSortedGroupsFromDb(tableName, attributeName));
+            return histogram;
+        }
+
+        internal async Task<IHistogram?> GetHistogramFromCacheOrNull(string tableName, string attributeName)
+        {
+            if (HistogramCacher.Instance == null)
+                return null;
+            string hash = await DataGatherer.GetTableAttributeColumnHash(tableName, attributeName);
+            var retVal = HistogramCacher.Instance.GetValueOrNull(new string[] { tableName, attributeName, hash, Depth.ToString() });
```
Comment (kris701) [inline Jantimizer/Histograms/Managers/MinDepthHistogramManager.cs:42]: Missing `typeof(HistogramMinDepth).Name` as well (Look at the `CacheHistogram()` function to see what strings are used in the hash)

**Review by kris701 — CHANGES_REQUESTED** [review]

File: Jantimizer/ExperimentSuite/experiments.json [hunk 2]
```diff
@@ -42,6 +42,48 @@
         "BucketSize": 10
       }
     },
+    {
+      "ExperimentName": "Experiment One_min",
```
Comment (jamadaha) [inline Jantimizer/ExperimentSuite/experiments.json:46]: Is this not experiment 3?

**Review by jamadaha — CHANGES_REQUESTED** [review]

File: Jantimizer/ExperimentSuite/experiments.json [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
       "PreRunData": [
         {
           "ConnectorName": "POSGRESQL",
           "ConnectorID": "EquiDepth",
           "TestFiles": [
-            "BasicTests_SetupTables"
+            "BasicTests_SetupTables",
+            "WorldHappiness_With_GDP_Setup",
+            "Census_2013_Setup",
+            "COVID_Funding_Setup"
```
Comment (kris701) [inline Jantimizer/ExperimentSuite/experiments.json:15]: I guess you just stole the issue i was working on :'(

Comment (kris701) [reply Jantimizer/ExperimentSuite/experiments.json:15]: Also i still think its a better idea to wait with merging all these, until the optimiser can actually *work* with them.

**Review by kris701 — DISMISSED** [review]

File: Jantimizer/ExperimentSuite/experiments.json [hunk 4]
```diff
@@ -89,123 +101,59 @@
...
+            "BasicTests_ClumpedNumberBlocks",
+            "WorldHappiness_With_GDP",
+            "Census_2013",
             "COVID_Funding"
           ]
         }
       ],
       "OptionalTestSettings": {
-        "BucketSize": 512
+        "BucketSize": 10
```
Comment (kris701) [inline Jantimizer/ExperimentSuite/experiments.json:156]: Please do set all the bucketsizes to be like 256 instead

**Review by kris701 — CHANGES_REQUESTED** [review]

**Review by kris701 — APPROVED** [review]

**Review by jamadaha — APPROVED** [review]
It runs

**Pull request merged**